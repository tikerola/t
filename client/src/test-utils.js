import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "./styles/theme";
import { Router } from "react-router-dom";

import { createMemoryHistory } from "history";

const history = createMemoryHistory();
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={lightTheme}>
      <Router history={history}>{children}</Router>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
