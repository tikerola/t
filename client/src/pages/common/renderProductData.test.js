import React from "react";
import { render } from "../../test-utils";
import { RenderProductData } from "./RenderProductData";

const productData = [
  {
    id: "f33561de3a864f951a",
    type: "jackets",
    name: "EWHHOP ROOM",
    color: ["blue"],
    price: 52,
    manufacturer: "reps",
    availability: "IN STOCK",
  },
  {
    id: "0e4772c827c4296592fbd",
    type: "jackets",
    name: "WEERLEP METROPOLIS RAPTOR",
    color: ["black"],
    price: 98,
    manufacturer: "reps",
    availability: "OUT OF STOCK",
  },
  {
    id: "6d39a08b3bcae88a67",
    type: "jackets",
    name: "DERWEER TYRANNUS BANG",
    color: ["purple"],
    price: 15,
    manufacturer: "abiplos",
    availability: "LESS THAN 10",
  },
];

describe("<RenderProductData />", () => {
  it("should match snapshot", () => {
    const component = render(<RenderProductData productData={productData} />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should render productData", () => {
    const component = render(<RenderProductData productData={productData} />);

    const title1 = component.queryByText("EWHHOP ROOM");
    const title2 = component.queryByText("WEERLEP METROPOLIS RAPTOR");
    const title3 = component.queryByText("DERWEER TYRANNUS BANG");

    expect(title1).toBeDefined();
    expect(title2).toBeDefined();
    expect(title3).toBeDefined();

    const m1 = component.queryAllByText("reps");
    const m2 = component.queryByText("abiplos");

    expect(m1[0]).toBeDefined();
    expect(m1[1]).toBeDefined();
    expect(m2).toBeDefined();

    const a1 = component.queryByText("IN STOCK");
    const a2 = component.queryByText("OUT OF STOCK");
    const a3 = component.queryByText("LESS THAN 10");

    expect(a1).toBeDefined();
    expect(a2).toBeDefined();
    expect(a3).toBeDefined();

    const p1 = component.queryByText("52");
    const p2 = component.queryByText("98");
    const p3 = component.queryByText("15");

    expect(p1).toBeDefined();
    expect(p2).toBeDefined();
    expect(p3).toBeDefined();

    const colors = component.queryAllByTestId("color-button");
    expect(colors).toHaveLength(3);
  });
});
