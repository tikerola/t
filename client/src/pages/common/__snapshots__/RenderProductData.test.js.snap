// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderProductData /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gKsewC colzyP"
  >
    <div
      class="sc-bdfBwQ kOQqDM"
    >
      <div
        class="sc-iBPRYJ hxjBfL"
      >
        EWHHOP ROOM
      </div>
      <div
        class="sc-fubCfw hOXhA"
      >
        reps
      </div>
      <div
        class="sc-kEjbxe jhNXHq"
      >
        Available colors: 
        <div
          class="sc-iqHYGH cyinRC"
          color="blue"
          data-testid="color-button"
        />
        <div
          class="sc-pFZIQ loYqnT"
        >
          IN STOCK
        </div>
      </div>
      <div
        class="sc-jrAGrp ebDMLb"
      >
        $52
      </div>
    </div>
    <div
      class="sc-bdfBwQ kOQqDM"
    >
      <div
        class="sc-iBPRYJ hxjBfL"
      >
        WEERLEP METROPOLIS RAPTOR
      </div>
      <div
        class="sc-fubCfw hOXhA"
      >
        reps
      </div>
      <div
        class="sc-kEjbxe jhNXHq"
      >
        Available colors: 
        <div
          class="sc-iqHYGH dNWqiX"
          color="black"
          data-testid="color-button"
        />
        <div
          class="sc-pFZIQ loYqnT"
        >
          OUT OF STOCK
        </div>
      </div>
      <div
        class="sc-jrAGrp gVzlQd"
      >
        $98
      </div>
    </div>
    <div
      class="sc-bdfBwQ kOQqDM"
    >
      <div
        class="sc-iBPRYJ hxjBfL"
      >
        DERWEER TYRANNUS BANG
      </div>
      <div
        class="sc-fubCfw hOXhA"
      >
        abiplos
      </div>
      <div
        class="sc-kEjbxe jhNXHq"
      >
        Available colors: 
        <div
          class="sc-iqHYGH eWAKNS"
          color="purple"
          data-testid="color-button"
        />
        <div
          class="sc-pFZIQ loYqnT"
        >
          LESS THAN 10
        </div>
      </div>
      <div
        class="sc-jrAGrp laVZyc"
      >
        $15
      </div>
    </div>
  </div>
</DocumentFragment>
`;
